.github/actions/publish-to-s3/action.yml

```diff
@@ -0,0 +1,107 @@
+name: Publish to S3
+description: Publishes the specified directory using aws s3 sync
+
+inputs:
+  AWS_REGION:
+    description: The region of your AWS bucket
+    default: ca-central-1
+    required: false
+  AWS_ROLE:
+    description: The AWS role to assume. This role must have write permission for your bucket
+    required: true
+  BUCKET_PATH:
+    description: The full path of your bucket, including subdirectories, to which to publish
+    required: true
+  BUILD_DIRECTORY:
+    description: The directory within your repo to publish to S3 (e.g. "./build/")
+    required: true
+  CACHE:
+    description: An optional comma-separated list of all file extensions you wish to have cached for 1 year (e.g. "js,css")
+    default: ""
+    required: false
+  CACHE_DEFAULT:
+    description: An optional default caching policy to apply to all files (e.g. "--cache-control 'max-age=120'")
+    default: ""
+    required: false
+  COMPRESS:
+    description: Whether to compress eligible files (if "true", will compress all of html/css/js/json/svg/xml)
+    default: false
+    required: false
+
+runs:
+  using: composite
+  steps:
+    - name: Assume role
+      uses: Brightspace/third-party-actions@aws-actions/configure-aws-credentials
+      with:
+        aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+        aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+        aws-session-token: ${{ secrets.AWS_SESSION_TOKEN }}
+        role-to-assume: ${{ inputs.AWS_ROLE }}
+        role-duration-seconds: 3600
+        aws-region: ${{ inputs.AWS_REGION }}
+
+    - name: Parse files
+      id: parse-files
+      env:
+        CACHE: ${{ inputs.CACHE }}
+        COMPRESS: ${{ inputs.COMPRESS }}
+      run: |
+        typesToCompress=()
+        [[ $COMPRESS != 'false' ]] && typesToCompress=("html" "css" "js" "json" "svg" "xml")
+        typesToCache=(${CACHE//,/ })
+
+        compressAndCache=""
+        compressOnly=""
+        cacheOnly=""
+        neither=""
+
+        for type in ${typesToCompress[@]}; do
+          if [[ " ${typesToCache[*]} " =~ " ${type} " ]]; then
+            compressAndCache+="--include '*.${type}' "
+          else
+            compressOnly+="--include '*.${type}' "
+          fi
+          neither+="--exclude '*.${type}' "
+        done
```

Comment:
This is super cool! If we're not going to use caching out of the gate though (which I agree it doesn't make sense here), is this maybe overly complex? I'd be sad to lose it though when/if this becomes not-internal, so maybe it's fine.

Reply:
Keeping it! Spent all that time finally getting it working 😆. Should be nice and general for the future.